eth/eth_test/common_test.go

```diff
@@ -0,0 +1,178 @@
+	"github.com/ethereum/go-ethereum/accounts/abi/bind"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/rpc"
+	"go.uber.org/zap/zaptest"
+	"math/big"
+	"net/http/httptest"
+	"strings"
+	"testing"
+)
+
+type CommonTestInput struct {
+	t             *testing.T
+	sim           *simulator.SimulatedBackend
+	boundContract *simcontract.Simcontract
+	blockNum      *uint64
+	nodeStorage   storage.Storage
+	doInOneBlock  bool
+}
+
+func NewCommonTestInput(
+	t *testing.T,
+	sim *simulator.SimulatedBackend,
+	boundContract *simcontract.Simcontract,
+	blockNum *uint64,
+	nodeStorage storage.Storage,
+	doInOneBlock bool,
+) *CommonTestInput {
+	return &CommonTestInput{
+		t:             t,
+		sim:           sim,
+		boundContract: boundContract,
+		blockNum:      blockNum,
+		nodeStorage:   nodeStorage,
+		doInOneBlock:  doInOneBlock,
+	}
+}
+
+type testEnv struct {
+	eventSyncer   *eventsyncer.EventSyncer
+	validators    []*testValidatorData
+	ops           []*testOperator
+	nodeStorage   storage.Storage
+	sim           *simulator.SimulatedBackend
+	boundContract *simcontract.Simcontract
+	auth          *bind.TransactOpts
+	shares        [][]byte
+	execClient    *executionclient.ExecutionClient
+	rpcServer     *rpc.Server
+	httpSrv       *httptest.Server
+}
+
+func setupEnv(
+	t *testing.T,
+	ctx context.Context,
+	testAddresses []*ethcommon.Address,
+	validatorsCount uint64,
+	operatorsCount uint64,
+) (*testEnv, error) {
+	logger := zaptest.NewLogger(t)
+
+	// Create operators RSA keys
+	ops, err := createOperators(operatorsCount, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	validators := make([]*testValidatorData, validatorsCount)
+	shares := make([][]byte, 10)
+
+	// Create validators, BLS keys, shares
+	for i := 0; i < 10; i++ {
+		validators[i], err = createNewValidator(ops)
+		if err != nil {
+			return nil, err
+		}
+
+		shares[i], err = generateSharesData(validators[i], ops, testAddrAlice, i)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	eh, validatorCtrl, nodeStorage, err := setupEventHandler(t, ctx, logger, ops[0], &testAddrAlice, true)
+	if err != nil {
+		return nil, err
+	}
+	if validatorCtrl == nil {
+		return nil, fmt.Errorf("error: validatorCtrl is empty")
+	}
+
+	// Adding testAddresses to the genesis block mostly to specify some balances for them
+	sim := simTestBackend(testAddresses)
+
+	// Create JSON-RPC handler
+	rpcServer, _ := sim.Node.RPCHandler()
+	// Expose handler on a test server with ws open
+	httpSrv := httptest.NewServer(rpcServer.WebsocketHandler([]string{"*"}))
+	addr := "ws:" + strings.TrimPrefix(httpSrv.URL, "http:")
+
+	parsed, _ := abi.JSON(strings.NewReader(simcontract.SimcontractMetaData.ABI))
+	auth, _ := bind.NewKeyedTransactorWithChainID(testKeyAlice, big.NewInt(1337))
+	contractAddr, _, _, err := bind.DeployContract(auth, parsed, ethcommon.FromHex(simcontract.SimcontractMetaData.Bin), sim)
+	if err != nil {
+		t.Errorf("deploying contract: %v", err)
```

Comment:
Should it be logged outside of the function? I feel like we could get rid of having `t` here, however, `setupEventHandler` uses it, so `setupEventHandler` and this function use two different approaches: The former asserts errors, the latter returns them. I'd align them to have similar approach to error handling

Reply:
Fixed